psycopg/pqpath.c

```diff
@@ -1735,6 +1739,7 @@ pq_read_replication_message(replicationCursorObject *repl, replicationMessageObj
         goto exit;
     }
 
+    repl->wal_end = wal_end;
```

Comment:
Doing this here seems inconsistent. The variable may be uninitialized. Shouldn't it be done in the `buffer[0] == 'w'` block?

Reply:
I think you get here only from `buffer[0] == 'w'` block and only if there was no error there. I'll move it inside the block to be more clear.